pkg/users/repository.go

```diff
@@ -0,0 +1,56 @@
+package users
+
+import (
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type UserRepository interface {
+	RegisterUser(user *User) error
+}
+
+type UserNeo4jRepository struct {
+	Driver neo4j.Driver
+}
+
+func (u *UserNeo4jRepository) RegisterUser(user *User) (err error) {
+	session, err := u.Driver.NewSession(neo4j.SessionConfig{
+		AccessMode: neo4j.AccessModeWrite,
```

Comment:
Is setting `AccessMode` necessary if you're then using an explicit `WriteTransaction` call with a transaction function?

Reply:
You're right, it is not necessary at all. Thanks for spotting that!

- `AccessModeWrite` is the default value 
 - the access mode configured at the session level is the default mode for the session, which is completely disregarded when explicit read or write transactions are used.